### PR TITLE
DM-31934: Replace HH:MM:SS conversion code with astropy equivalents.

### PR DESCRIPTION
**Pull request by ebellm**

### CODE REVIEW

File: python/lsst/ap/association/skyBotEphemerisQuery.py [hunk 1]
```diff
@@ -231,10 +233,12 @@ def _skybotConeSearch(self, expCenter, epochJD, queryRadius):
...
             coldict = dict(zip(dfSSO.columns, columns))
             dfSSO.rename(columns=coldict, inplace=True)
-            # TODO: DM-31934 Replace custom conversion code with
-            #       Astropy.coordinates.
-            dfSSO["ra"] = self._rahms2radeg(dfSSO["RA(h)"])
-            dfSSO["decl"] = self._decdms2decdeg(dfSSO["DE(deg)"])
+            # Data returned in hourangle format.
+            radec = dfSSO["RA(h)"]+' '+dfSSO["DE(deg)"]
+            skycoo = SkyCoord(radec.values, unit=(u.hourangle, u.deg))
+            # Use astropy SkyCoord to convert to decimal degrees.
```
Comment (kfindeisen) [inline python/lsst/ap/association/skyBotEphemerisQuery.py:239]: This comment would make more sense before the line that uses `SkyCoord`.

**Review by kfindeisen — APPROVED** [review]
Looks good, style comments only.

File: python/lsst/ap/association/skyBotEphemerisQuery.py [hunk 2]
```diff
@@ -231,10 +233,12 @@ def _skybotConeSearch(self, expCenter, epochJD, queryRadius):
             columns = [col.strip() for col in dfSSO.columns]
             coldict = dict(zip(dfSSO.columns, columns))
             dfSSO.rename(columns=coldict, inplace=True)
-            # TODO: DM-31934 Replace custom conversion code with
-            #       Astropy.coordinates.
-            dfSSO["ra"] = self._rahms2radeg(dfSSO["RA(h)"])
-            dfSSO["decl"] = self._decdms2decdeg(dfSSO["DE(deg)"])
+            # Data returned in hourangle format.
+            radec = dfSSO["RA(h)"]+' '+dfSSO["DE(deg)"]
+            skycoo = SkyCoord(radec.values, unit=(u.hourangle, u.deg))
```
Comment (timj) [inline python/lsst/ap/association/skyBotEphemerisQuery.py:238]: Isn't it a bit overkill to use `SkyCoord` here?

```python
from astropy.coordinates import Angle

dfSSO["ra"] = Angle(dfSSO["RA(h)"])
dfSSO["decl"] = Angle(dfSSO["DE(deg)", unit="deg")
```
eg.:
```python
>>> c.Angle("5 32 3.4", unit="deg").deg
5.534277777777778
>>> c.Angle("5 32 3.4", unit="h").deg
83.01416666666665
```

Comment (kfindeisen) [reply python/lsst/ap/association/skyBotEphemerisQuery.py:238]: I would have used `Angle` myself, yes, but the most important thing is to have an off-the-shelf solution.

Comment (timj) [reply python/lsst/ap/association/skyBotEphemerisQuery.py:238]: SkyCoord adds an extra layer of indirection and requires you to concatenate the two strings. I agree that anything is better than doing it ourselves.

Comment (ebellm) [reply python/lsst/ap/association/skyBotEphemerisQuery.py:238]: The optimization here seems pretty tiny--we're going to throw this whole module away--but sure, that works.

**Pull request merged**